.script/tests/detectionTemplateSchemaValidation/Models/ModelValidationAttributes/NewEntityMappingsAttribute.cs

```diff
@@ -43,6 +43,13 @@ protected override ValidationResult IsValid(object value, ValidationContext vali
                 {
                     return new ValidationResult($"An old mapping for entity '{oldEntityMappingColumnName}' does not have a matching new mapping entry.");
                 }
+                if (HasAMatchingNewMappingEntry(template, oldEntityMappingColumnName))
```

Comment:
Nice check
In fact we can widen this to check that all the column names mapped in the new entity mappings are present in the final projected columns of the query, but this won't work all the time since we don't have the convention of projecting the columns we use.

but this is still good for maintaining alignment with the mappings of legacy entities.

worth mentioning that the legacy entity mappings are not required anymore, so new templates can name the columns freely and not using the XXCustomEntity suffix

Reply:
@laithhisham can you please review this PR